internal/session/session.go

```diff
@@ -0,0 +1,89 @@
+package session
+
+import (
+	"github.com/dgraph-io/ristretto"
+	"github.com/google/uuid"
+	"github.com/neicnordic/sda-download/internal/config"
+	log "github.com/sirupsen/logrus"
+)
+
+// SessionCache stores dataset permission lists
+var SessionCache *ristretto.Cache
+
+// DatasetCache stores the dataset permissions
+// and information whether this information has
+// already been checked or not. This information
+// can then be used to skip the time-costly
+// authentication middleware
+// DatasetCache==nil, session doesn't exist
+// DatasetCache.Datasets==nil, session exists, user has no permissions (this case is not used in middleware.go)
+// DatasetCache.Datasets==[]string{}, session exists, user has permissions
+type DatasetCache struct {
+	Datasets []string
+}
+
+// InitialiseSessionCache creates a cache manager that stores keys and values in memory
+func InitialiseSessionCache() (*ristretto.Cache, error) {
+	log.Debug("creating session cache")
+	sessionCache, err := ristretto.NewCache(
+		&ristretto.Config{
+			NumCounters: 1e7,     // Num keys to track frequency of (10 M = max. 1 M users)
+			MaxCost:     1 << 30, // Maximum cost of cache (1 GiB)
+			BufferItems: 64,      // Number of keys per Get buffer
+		},
+	)
+	if err != nil {
+		log.Errorf("failed to create session cache, reason=%v", err)
+		return nil, err
+	}
+	log.Debug("session cache created")
+	return sessionCache, nil
+}
+
+// Get returns a value from cache at key
+// DON'T USE DEBUG LOGS IN PRODUCTION, AS SESSION KEY WILL BE LOGGED
```

Comment:
I don't think we need to log it even for debug. We might get an alert as: https://github.com/neicnordic/sda-pipeline/security/code-scanning/3?query=ref%3Arefs%2Fheads%2Fmaster so we can just remove that and add it when we develop

Reply:
removed